### ISSUE TITLE
[user.js] Updating extensions page

### ISSUE THREAD

**Issue by TechnologyClassroom** [arkenfox/user.js#1573]
I see your page [4.1-Extensions](https://github.com/arkenfox/user.js/wiki/4.1-Extensions) linked in privacy threads on Reddit and other GH issues often.  I want to point out two things that could be updated.

- Fingerprinting
  - The page focuses on CanvasBlocker and RFP for anti-fingerprinting.  [JShelter](https://jshelter.org/) is another option that I believe is a great addition to this list.  JShelter approaches anti-fingerprinting in several ways beyond what CanvasBlocker can do.  Edit: the other JShelter issue is set to "This conversation has been locked as resolved and limited to collaborators."
- Dismissal of NoScript.
  - Regarding NoScript, the page says, `Redundant with uBlock Origin` which is not true.  uBlock Origin has many lists that stop many anti-patterns, but their jobs are different and the two extensions work well together.  NoScript blocks JavaScript from domains and third-party domains outright.

**Comment by rusty-snake** [comment]
> JShelter

There's a search function: #1388

**Comment by rusty-snake** [comment]
>  NoScript blocks JavaScript from domains and third-party domains outright.

Guess what uB can do.

**Comment by TechnologyClassroom** [comment]
uBlock Origin can do it, but does not do it by default.  The UI is mostly geared towards enabling or disabling lists and then turning the entire extension off when it breaks a site.  The UI of NoScript blocks JS outright and then is based around allowing domains and third-party domains that the user trusts.

**Comment by rusty-snake** [comment]
Did you read the documentation of uB linked on the extension page?

**Comment by TechnologyClassroom** [comment]
Yes, but consider how many clicks it takes for a user to install two extensions vs configuring uBlock Origin to hard mode.  The more configuring required from the user, results in people going half-way and being less private.  The documentation for uBlock Origin advanced configuration could say that if you do all of this, then NoScript is not necessary.

**Comment by TechnologyClassroom** [comment]
The JShelter issue is closed and locked so no one else can talk about it.

With NoScript, I often see people link to the page saying it is redundant without specifying the documentation about hardening uBlock Origin.  I think this is actively worse because the line that I highlighted in the issue provides a false sense of security that a user is protected against all of JS when they are likely only using the default configuration of uBlock Origin.  A false sense of security is often times worse than no security at all.

**Comment by fxbrit** [comment]
> The JShelter issue is closed and locked

that's because pants decided it's not going to make the list, but before locking and closing there was a rather long discussion that explains why the decision was taken. I see no reason to further discuss it in other issues honestly..

> I often see people link to the page saying it is redundant without specifying the documentation about hardening uBlock Origin

as rusty said there's a note about uBO's modes right in the wiki, it's the first buller point. if people are too lazy to click it's up to them, and if instead users are deciding medium mode is too much, that's also a respectable choice.

I also don't buy into the narrative of disabling all JS by default: it's extreme and it's not usable. I struggle to find a use case that requires such measures on a general purpose browser (or any browser really).

**Comment by TechnologyClassroom** [comment]
The JShelter discussion was long, but the discussion was between you all and someone who read about the extension and tested it with one page.  It was also mostly side-tracked about Tor browser.  Did you ask your questions to the JShelter developers?  What tests did you run on JShelter?

NoScript isn't about using no JavaScript.  It is about blocking by default and then allowing only what needs to be allowed.

JShelter works well with uBlock Origin and NoScript.  Once a script is allowed by uBlock Origin and NoScript, data is then guarded by alterations to the API calls.

**Comment by fxbrit** [comment]
https://github.com/arkenfox/user.js/issues/1388#issuecomment-1057542283, https://github.com/arkenfox/user.js/issues/1388#issuecomment-1057959990 and https://github.com/arkenfox/user.js/issues/1388#issuecomment-1057989887 contain hints. the approach of JShelter is not in line with Arkenfox's wiki: this is fine of course, devs can do whatever they want, but at the same time it's clear why it's not going to be recommended. did you read the wiki? 

also I don't mean to be rude, but I don't understand what you would expect me (or pants I think, since I'm a collaborator) to ask to JShelter devs? Firefox has RFP, we believe in it, we trust it, test it and use it. this seems like a reasonable stance.

> What tests did you run on JShelter?

based on the reply I assume it was TZP + reading the values and analyzing them.

>  It is about blocking by default and then allowing only what needs to be allowed.

as I said above, uBO can do the same.

**Comment by Thorin-Oakenpants** [comment]
So what are you trying to achieve? And let's be clear, what we recommend is for the masses, as a default. You are free to do as you like. Also note, that the wiki says

> This list covers privacy and security related extensions only. While we believe these are the very best of the best, this can be subjective depending on your needs. We are also not saying you have to use all these extensions

### noscript

> uBlock Origin can do it, but does not do it by default.

Your argument is that NS disables JS by default. Why? JS is not evil per se and does not fit this repo's threat model. If you are this concerned, you should be using Tor Browser on safest (or you know, use uBO in whatever hard mode/config you like).

Think about that for a second, as a default, no JS. It would make the internet quite unusable, and users would have to add lots of exceptions (YMMV) etc (and I find the UI "quite" unintuitive TBH, but ignore that). Users thus need to be technically advanced enough to use it - to know what is causing an issue, the ramifications, etc. Using the web should be easy. not constantly fighting it and training an extension.

This makes your claim that uBO requires changes to be more hardened ... moot to say the least. Seriously, arguing that it takes a few settings/clicks to setup hardened mode in uBO is counterintuitive, when hardened mode also indicates the user is advanced enough to use it, and hypercritical to the max when NS also requires advanced users

> The UI is mostly geared towards enabling or disabling lists

The settings UI, which the user has to explicitly open, has a tab for lists, but there are tabs for other things. The lists are only part of it. The toolbar icon's popup panel on the other is visually accessible all the time, and totally geared to adding/changing rules - we're assuming users who are advanced, remember (because NoScript requires advanced and you're trying to compare NS to uBO)

tl;dr: uBO is recommended because it is a content blocker. The lists are great, as a start. If users are capable, then they can turn on advanced mode, config how they like (there is no one size fits all) - if they can do this, they can do anything NS does (within reason: I am not here to cherry pick specific items) - i.e they can achieve the same result (hardened but usable).

The default is easy to use (uBO) because using the web should be easy ... through to whatever you like (fight every fucking request for all I care - but that's _your_ decision) - I use uBO in hard mode and I find it super easy and I'm not fighting _anything_, I barely change anything in it, everything works _for me_. But that's _ME_, not the default user

### fingerprinting

nah. not going to waste too much time on this, sorry.

As someone with extensive knowledge in this field, the answer is that extensions are not the answer. RFP is the best solution for FF, and _if_ you can't handle that, then an extension is the next best thing - and we promote canvas blocker due to the developer (kkapsner's record and knowledge) and we only recommend that canvas, webgl if used perhaps, and maybe audio be used - this is all you need. And that's only if you care about possible fingerprinting in Firefox.

Could you use a different extension? Do what you like. As pointed out before, I do not find JShelter to be worthy in any regard - holy fuck, it even leaks specific unique changes in prototype navigator, it fucks with audio for no real gain. It fucks with timing (doing this is dodgy to say the least IMO). Use Cydec, holy fuck it messes with `Math.` which is insane. Use Trace, holy fuck that is such a PoS. And so on. I have spent year(s) exploiting extension BS.

Although I personally think CB has a little feature creep and a few items not really needed, it's as solid as you can get.

If you really care about FPing, use Tor Browser.

**Comment by Thorin-Oakenpants** [comment]
> and someone who read about the extension and tested it with one page

that someone was me, and that _one_ page was TZP, which has quite some extensive "smarts". These smarts are to visually show information paradoxes, unexpected errors, weird results - and basically unmask BS. It doesn't _have_ to do that. The data is already in the metrics collected - prototype lies, proxy lies, errors, enumeration of keys/properties after constructor, FP methods (such as randomizing per execution or static, amount of randomizing - both in e.g. canvas, domrect methods, etc), and so on.

however, if I wanted to collect fingerprints, or compare them: to ascertain what needs to be protected in RFP and Tor Browser, then I also want to filter out any BS - which is the purpose of the test itself.

that one page and one person is enough. Or do you think I'm not qualified?

**Comment by Thorin-Oakenpants** [comment]
> it fucks with audio for no real gain

I made a pun. damn, I'm good

**Comment by rusty-snake** [comment]
NoScript does not resolve CNAMEs AFAIK. You can tell uB to not resolve CNAMEs if it is a problem for you.

Instead of allowing specific domains, you can also go to easy mode on a site.

Nooping common CDNs globally solves also a few rounds of nooping.

**Comment by Thorin-Oakenpants** [comment]
this is not a forum to ask how to use uBO - use reddit
